canonical-web-and-design/ubuntu.com
# Switch API environment depending on query param

## Done

- Added checks for the query params. if the url contains `?ua_staging=true` we use staging values for both Stripe and Contracts API if not we default to live ones.

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/advantage/subscribe?test_backend=true
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- check that the purchase works with a test card (4242424242...)
- Go to http://0.0.0.0:8001/advantage/subscribe without the query param
- Try to complete a purchase using a test card and see that it fails with an error

## Issue / Card

Fixes #8828
